server/tasks/process_file.py

```diff
@@ -74,11 +70,17 @@ async def async_process_streaming_output(
             data = await consumer.getmany(timeout_ms=3000, max_records=batch_size)
             for tp, messages in data.items():
                 if messages:
-                    result_handler(messages)
+                    logger.info(f"Handling {messages=} in topic {tp.topic}")
+                    data = [msg.value for msg in messages]
+                    result_handler(data)
+                    logger.info(f"Handled {len(messages)} messages in topic {tp.topic}")
                 else:
                     logger.info(f"No messages in topic {tp.topic}")
+            if not data:
+                logger.info(f"No messages in any topic")
         finally:
             job = process_file_streaming.AsyncResult(input_job_id)
+            # TODO no way to recover here if connection to main app is lost, job will be stuck at "PENDING" so this will loop forever
```

Comment:
naive question: How would this manifest, if we lose the connection? Can we do something like wrap the `job = process_file_streaming.AsyncResult(input_job_id)` in a try/except, and if we hit an error when trying to get the job, assume the job failed?

Reply:
The problem is that celery will report all nonexistent or unknown job ids as `Pending`. There's no such thing as an error trying to get the job besides that status. This is a problem with using celery as our status mechanism in general.

Reply:
ugh, gotcha